docs/source/JAX_for_LLM_pretraining.ipynb

```diff
@@ -95,11 +98,27 @@
     "id": "Rcji_799n4eA"
    },
    "source": [
-    "**Note:** If you are using [Google Colab](https://colab.research.google.com/), select the free Google Cloud TPU v2 as the hardware accelerator.\n",
+    "**Note:** If you are using [Kaggle](https://www.kaggle.com/), select the free TPU v5e-8 as the hardware accelerator. If you are using [Google Colab](https://colab.research.google.com/), select the free Google Cloud TPU v5e-1 as the hardware accelerator. You may also use Google Cloud TPUs.\n",
     "\n",
     "Check the available JAX devices, or [`jax.Device`](https://jax.readthedocs.io/en/latest/_autosummary/jax.Device.html), with [`jax.devices()`](https://jax.readthedocs.io/en/latest/_autosummary/jax.devices.html). The output of the cell below will show a list of 8 (eight) devices."
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "if os.path.exists('/content/'):\n",
+    "  platform = \"Colab\"\n",
```

Comment:
What do you think of checking `jax.device_count()` and using that instead of platform, e.g. `if jax.device_count() == 8` instead of `if platform == "Kaggle"`? 

(The CI jobs won't be running on either Colab or GCP I think, and maybe people want to play around with it on a local cpu runtime e.g. And for defining the mesh, etc. I think device_count is what we care about.)

Reply:
Good idea!